src/babel/transformation/transformers/es6/classes/vanilla.js

```diff
@@ -592,11 +592,16 @@ export default class ClassTransformer {
   pushInherits() {
     if (!this.isDerived || this.pushedInherits) return;
 
-    this.pushedInherits = true;
-    this.body.push(t.expressionStatement(t.callExpression(
+    var inherits = t.expressionStatement(t.callExpression(
       this.file.addHelper("inherits"),
       [this.classRef, this.superName]
-    )));
+    ));
+    // Ensure that the constructor inheritance is set up before any properties
+    // can be assigned to the prototype.
+    inherits._blockHoist = 2;
```

Comment:
You should be able to just do `this.body.unshift` rather than using `_blockHoist`.